### PR TITLE
[lambda][alert] adding support for sns as a source

### PR DESCRIPTION
**Pull request by ryandeivert**
to @airbnb/streamalert-maintainers 
size: small

work in progress for #64 
## changes ##
* Adding sns source support
* Updating tests to include sns sources

### CODE REVIEW

**Comment by jacknagz** [comment]
anotha commit
anothaone

File: stream_alert/rule_processor/config.py [hunk 1]
```diff
@@ -67,8 +67,8 @@ def validate_config(config):
 
         # check sources attributes
         elif config_key == 'sources':
-            if not set(settings.keys()).issubset(set(['kinesis', 's3'])):
-                raise ConfigError('Sources missing kinesis or s3 keys')
+            if not set(settings.keys()).issubset(set(['kinesis', 's3', 'sns'])):
```
Comment (jacknagz) [inline stream_alert/rule_processor/config.py]: instead of doing `set([])` let's just do `{}`

File: stream_alert/rule_processor/classifier.py [hunk 2]
```diff
@@ -130,11 +130,14 @@ def map_source(self, payload):
             payload.service = 'kinesis'
         elif 's3' in payload.raw_record:
             payload.service = 's3'
+        elif 'Sns' in payload.raw_record:
+            payload.service = 'sns'
 
         # map the entity name from a record
         entity_mapper = {
```
Comment (jacknagz) [inline stream_alert/rule_processor/classifier.py]: #simplify (discussed offline)

File: stream_alert/rule_processor/handler.py [hunk 3]
```diff
@@ -86,6 +88,12 @@ def s3_process(self, payload, classifier):
             classifier.classify_record(payload, data)
             self.process_alerts(payload)
 
+    def sns_process(self, payload, classifier):
+        """Process SNS data for alerts"""
+        data = StreamPreParsers.pre_parse_sns(payload.raw_record)
```
Comment (jacknagz) [inline stream_alert/rule_processor/handler.py]: this method looks almost the same as the kinesis pre parser, can we simplify?

Comment (ryandeivert) [inline stream_alert/rule_processor/handler.py]: I made a small change, but it just mostly consolidates duplicate code into a method. The `pre_parse..` methods pull data from different keys, so this is about as simple as this get (see `pre_parsers.py` for more context).

**Pull request merged**